app/main.py

```diff
@@ -1,2 +1,123 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+
+    def __setitem__(
+            self,
+            key: int | float | str | bool | tuple | object,
+            value: Any
+    ) -> None:
+
+        hash_cell = hash(key) % len(self.hash_table)
+
+        if (
+                self.hash_table[hash_cell]
+                and self.hash_table[hash_cell].key == key
+        ):
+            self.hash_table[hash_cell].value = value
+            return
+        if (
+                self.hash_table[hash_cell]
+                and self.hash_table[hash_cell].key != key
+        ):
+            for item in self.hash_table:
+                if item and item.key == key:
+                    item.value = value
+                    return
+        hash_cell = self.get_free_cell(hash_cell)
+        self.add_new_item(hash_cell, key, value)
+
+    def __getitem__(self, key: Any) -> Any:
```

Comment:
```suggestion
    def __getitem__(self, key: Hashable) -> Any:
```

please fix everywhere